### PR TITLE
Made DAOs in API independent/self-sufficient

### LINKED ISSUE #746: Create a Data Access Object for Event
This will make it consistent with other service models . Event's DAO will have the same methods as other DAO. 

This is also required to proceed #728

### PR DESCRIPTION
**Pull request by aviaryan**
Fixes #746 and #747 

This PR makes the DAO independent meaning if you have the params of a request, you can now use the DAO to successfully execute any type of request and get the return value.
So it will add the ability to internally call an API and execute any type of request.

### CODE REVIEW

**Comment by aviaryan** [comment]
@shivamMg please review

File: open_event/api/events.py [hunk 1]
```diff
@@ -38,14 +38,59 @@
...
 
-# Helper function to fix datetime event payload
-# TODO: Make ServiceDAO more versatile so that event can use it
-def fix_payload(data):
-    data['start_time'] = EVENT_POST['start_time'].from_str(data['start_time'])
-    data['end_time'] = EVENT_POST['end_time'].from_str(data['end_time'])
-    data['closing_datetime'] = EVENT_POST['closing_datetime'].from_str(
-        data['closing_datetime'])
-    return data
+class EventDAO(ServiceDAO):
```
Comment (shivamMg) [inline open_event/api/events.py]: It would be better to create a BaseDAO, and then extend it to EventDAO and ServiceDAO. What do you say?

Comment (aviaryan) [inline open_event/api/events.py]: Sounds good. BaseDAO can be used for a parent-level (or independent) model.

**Comment by shivamMg** [comment]
@aviaryan Nice work.

**Pull request merged**